### ISSUE TITLE
Underscore variable $_ should be considered undefined in POSIX sh

### ISSUE THREAD

**Issue by XSven** [koalaman/shellcheck#1647]
#### For bugs
- Rule Id: SC2039 
- My shellcheck version: online

#### For new checks and feature suggestions
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/sh

[ "$_" -eq 1 ]

```
#### Here's what I wanted or expected to see:
According to the rational section on [Shell Variables](http://pubs.opengroup.org/onlinepubs/9699919799/xrat/V4_xcu_chap02.html#tag_23_02_05_03) the underscore variable should be undefined in POSIX sh. I would expect a complain refering to the rule id SC2039, if the underscore variable is used?!

**Comment by koalaman** [comment]
ShellCheck essentially ignored the variable name `_` because it's frequently used as a throwaway variable, e.g. in `read -r lat lng _ user _`. This did not account for the case where it's referenced, presumably to mean "the last argument of the previous command". You now get a warning in POSIX sh. Thanks!
